trustpoint/devices/models.py

```diff
@@ -86,13 +86,19 @@ def revoke_ldevid(self: Device, revocation_reason) -> bool:
         if not self.ldevid:
             return False
 
-        if self.device_onboarding_status == Device.DeviceOnboardingStatus.ONBOARDED:
-            self.device_onboarding_status = Device.DeviceOnboardingStatus.REVOKED
-
-        self.ldevid.revoke(revocation_reason)
+        revocation_success = self.ldevid.revoke(revocation_reason)
         self.ldevid = None
+        if self.device_onboarding_status == Device.DeviceOnboardingStatus.ONBOARDED:
+            if revocation_success:
+                self.device_onboarding_status = Device.DeviceOnboardingStatus.REVOKED
+            else:
+                # TODO(Air): Check if this makes sense to express the state "cannot revoke since CA is gone"
+                self.device_onboarding_status = Device.DeviceOnboardingStatus.ONBOARDING_FAILED
         self.save()
 
+        if not revocation_success:
+            log.error('Failed to revoke LDevID for device %s', self.device_name)
```

Comment:
Maybe move to line 96?

Reply:
Thank you so much for the approving review :)

I would not move this log, as in l. 96 the failed revocation would only be logged if the device onboarding status was `ONBOARDED`. There may be situations where the device has an LDevID, but not status onboarded (e.g. the onboarding failed after the certificate was issued)